webapp/channels/src/components/announcement_bar/text_dismissable_bar.tsx

```diff
@@ -20,67 +20,57 @@ interface Props extends Partial<AnnouncementBarProps> {
     className?: string;
 }
 
-type State = {
-    dismissed: boolean;
-}
-
-export default class TextDismissableBar extends React.PureComponent<Props, State> {
-    constructor(props: Props) {
-        super(props);
-
-        this.state = {
-            dismissed: true,
-        };
-    }
+const TextDismissableBar = (props: Props) => {
```

Comment:
Let's expand the props in the argument list, one prop by line.

```suggestion
const TextDismissableBar = ({
    allowDismissal,
    text,
    ...extraProps,
}: Props) => {
```